cypress/integration/11-room-directory/room-directory.spec.ts

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { SynapseInstance } from "../../plugins/synapsedocker";
+import { MatrixClient } from "../../global";
+
+describe("Room Directory", () => {
+    let synapse: SynapseInstance;
+
+    beforeEach(() => {
+        cy.startSynapse("default").then(data => {
+            synapse = data;
+
+            cy.initTestUser(synapse, "Ray");
+            cy.getBot(synapse, "Paul").as("bot");
+        });
+    });
+
+    afterEach(() => {
+        cy.stopSynapse(synapse);
+    });
+
+    it("should allow admin to add alias & publish room to directory", () => {
+        cy.window({ log: false }).then(win => {
+            cy.createRoom({
+                name: "Gaming",
+                preset: win.matrixcs.Preset.PublicChat,
+            }).as("roomId");
+        });
+
+        cy.viewRoomByName("Gaming");
+        cy.openRoomSettings();
+
+        // First add a local address `gaming`
+        cy.get(".mx_SettingsFieldset").contains(".mx_SettingsFieldset", "Local Addresses").within(() => {
+            cy.get(".mx_Field input").type("gaming");
+            cy.get(".mx_AccessibleButton").contains("Add").click();
```

Comment:
If there can be more accessible buttons on the page it would be better to do it this way:

```suggestion
            cy.contains(".mx_AccessibleButton", "Add").click();
```

Otherwise, as far as I understand, if Cypress finds any `.mx_AccessibleButton` it is going to wait for the `Add` content. works if there is no accessible button before. Could make the test more stable…

May apply to other places as well.